src/pages/auth/signup/signup.js

```diff
@@ -0,0 +1,130 @@
+import React, { useState } from "react";
+import { Link,useNavigate } from "react-router-dom";
+import { signupImage } from "../../../assets/images";
+import { AuthContainer } from "../components/authContainer";
+import {
+  BsFillEyeFill,
+  FaEnvelope,
+  AiOutlineArrowRight,
+} from "../../../assets/icons";
+import { useAuth } from "../../../context/auth/authContext";
+
+
+const Signup = () => {
+  const [formData, setFormData] = useState({
+    firstName: "",
+    lastName: "",
+    email: "",
+    password: "",
+  });
+  const { signup } = useAuth();
+  const navigate = useNavigate();
+  const handleOnChange = (e) => {
+    setFormData({...formData ,[e.target.id] : e.target.value});
+  };
+  return (
+    <AuthContainer title="Signup" imageUrl={signupImage}>
+      <form className="signup-form-container flex-col">
+        <ul>
+          <li className="list-style-none text-xs">
+            <div className="email-field-container">
+              <label className="text-xs" htmlFor="firstName">
+                First name
+              </label>
+              <input
+                id="firstName"
+                className="input-field form-control text-xs"
+                type="text"
+                placeholder="priya"
+                value={formData.firstName}
+                onChange={
+                  handleOnChange
```

Comment:
fix: You are using e in your handle change function, but you are not passing e from here